catalogue/tests/test_catalogue.py

```diff
@@ -121,6 +121,24 @@ def test_entry_points():
     assert "bar" in test_registry
 
 
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="Test does not support <3.10 importlib_metadata API")
+def test_entry_points_newer():
+    # Create a new EntryPoint object by pretending we have a setup.cfg and
+    # use one of catalogue's util functions as the advertised function
+    ep = catalogue.importlib_metadata.EntryPoint("bar", "catalogue:check_exists", "test_foo")
+    catalogue.AVAILABLE_ENTRY_POINTS["test_foo"] = catalogue.importlib_metadata.EntryPoints([ep])
+    assert catalogue.REGISTRY == {}
+    test_registry = catalogue.create("test", "foo", entry_points=True)
+    entry_points = test_registry.get_entry_points()
+    assert "bar" in entry_points
+    assert entry_points["bar"] == catalogue.check_exists
+    assert test_registry.get_entry_point("bar") == catalogue.check_exists
+    assert catalogue.REGISTRY == {}
+    assert test_registry.get("bar") == catalogue.check_exists
+    assert test_registry.get_all() == {"bar": catalogue.check_exists}
+    assert "bar" in test_registry
```

Comment:
Since these steps are duplicated in `test_entry_points_older()`, we could move both into a separate function.